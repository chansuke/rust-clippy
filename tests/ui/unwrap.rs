// compile-flags: --test

#![warn(clippy::unwrap_used)]

fn unwrap_option() {
    let opt = Some(0);
    let _ = opt.unwrap();
}

fn unwrap_result() {
    let res: Result<u8, ()> = Ok(0);
    let _ = res.unwrap();
}

#[cfg(test)]
mod test {
    #[test]
    fn test_flag() {
        let opt = Some(0);
        let _ = opt.unwrap();
    }
}

fn main() {}
