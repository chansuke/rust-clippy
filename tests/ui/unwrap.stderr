error: used `unwrap()` on `an Option` value
  --> $DIR/unwrap.rs:7:13
   |
LL |     let _ = opt.unwrap();
   |             ^^^^^^^^^^^^
   |
   = note: `-D clippy::unwrap-used` implied by `-D warnings`
   = help: if you don't want to handle the `None` case gracefully, consider using `expect()` to provide a better panic message

error: used `unwrap()` on `a Result` value
  --> $DIR/unwrap.rs:12:13
   |
LL |     let _ = res.unwrap();
   |             ^^^^^^^^^^^^
   |
   = help: if you don't want to handle the `Err` case gracefully, consider using `expect()` to provide a better panic message

error: used `unwrap()` on `an Option` value
  --> $DIR/unwrap.rs:20:17
   |
LL |         let _ = opt.unwrap();
   |                 ^^^^^^^^^^^^
   |
   = help: if you don't want to handle the `None` case gracefully, consider using `expect()` to provide a better panic message

error: aborting due to 3 previous errors

